hammer/technology/sky130/__init__.py

```diff
@@ -194,80 +585,103 @@ def setup_io_lefs(self) -> None:
         with open(source_path, 'r') as sf:
             with open(dest_path, 'w') as df:
                 self.logger.info("Modifying IO LEF: {} -> {}".format
-                    (source_path, dest_path))
+                                 (source_path, dest_path))
                 sl = sf.readlines()
                 for net in ['VCCD1', 'VSSD1']:
-                    start = [idx for idx,line in enumerate(sl) if 'PIN ' + net in line]
-                    end = [idx for idx,line in enumerate(sl) if 'END ' + net in line]
+                    start = [idx for idx, line in enumerate(
+                        sl) if 'PIN ' + net in line]
+                    end = [idx for idx, line in enumerate(
+                        sl) if 'END ' + net in line]
                     intervals = zip(start, end)
                     for intv in intervals:
-                        port_idx = [idx for idx,line in enumerate(sl[intv[0]:intv[1]]) if 'PORT' in line]
+                        port_idx = [idx for idx, line in enumerate(
+                            sl[intv[0]:intv[1]]) if 'PORT' in line]
                         for idx in port_idx:
-                            sl[intv[0]+idx]=sl[intv[0]+idx].replace('PORT', 'PORT\n      CLASS CORE ;')
+                            sl[intv[0]+idx] = sl[intv[0] +
+                                                 idx].replace('PORT', 'PORT\n      CLASS CORE ;')
                 for cell in [
                     'sky130_ef_io__connect_vcchib_vccd_and_vswitch_vddio_slice_20um',
                     'sky130_ef_io__disconnect_vccd_slice_5um',
                     'sky130_ef_io__disconnect_vdda_slice_5um',
                 ]:
                     # force class to spacer
-                    start = [idx for idx, line in enumerate(sl) if f'MACRO {cell}' in line]
-                    sl[start[0] + 1] = sl[start[0] + 1].replace('AREAIO', 'SPACER')
+                    start = [idx for idx, line in enumerate(
+                        sl) if f'MACRO {cell}' in line]
+                    sl[start[0] + 1] = sl[start[0] +
+                                          1].replace('AREAIO', 'SPACER')
 
                 # Current version has two one-off error that break lef parser.
-                self.logger.info("Fixing broken sky130_ef_io__analog_esd_pad LEF definition.")
-                start_broken_macro_list = ["MACRO sky130_ef_io__analog_esd_pad\n", "MACRO sky130_ef_io__analog_pad\n"]
-                end_broken_macro_list = ["END sky130_ef_io__analog_pad\n", "END sky130_ef_io__analog_noesd_pad\n"]
-                end_fixed_macro_list = ["END sky130_ef_io__analog_esd_pad\n", "END sky130_ef_io__analog_pad\n"]
+                self.logger.info(
+                    "Fixing broken sky130_ef_io__analog_esd_pad LEF definition.")
+                start_broken_macro_list = []
+                    #"MACRO sky130_ef_io__analog_esd_pad\n", "MACRO sky130_ef_io__analog_pad\n"]
```

Comment:
these don't seem to exist in the io spice file (causing this to error), is that correct?

Reply:
No, from what I can tell, the reason why that's there is because the END MACRO <macro_name> doesn't match the macro name declared in the beginning of the the MACRO block.

Reply:
oh is that why this breaks the parsing? it seems to run fine as it is here

Reply:
turns out the instructional machine lef is fixed. will remove the hack